include/base/libmesh_common.h

```diff
@@ -596,6 +596,15 @@ inline Tnew libmesh_cast_int (Told oldvar)
 #define libmesh_final
 #endif
 
+// Define backwards-compatible fallthrough attribute.
+// We should add support for standard C++17 [[fallthrough]]
+// Maybe [[clang::fallthrough]] or [[gcc::fallthough]] too?
+#if defined(__GNUC__) && (__GNUC__ > 6)
+#define libmesh_fallthrough __attribute__((fallthrough))
+#else
+#define libmesh_fallthrough
```

Comment:
Two questions: 
1. No parentheses (to match the look of `libmesh_error_msg()` et al)?
1. This will expand to an empty line with a semi-colon on most compilers. Won't that also trigger a warning? Should we use some non-statement like:
       
       #define libmesh_assert_msg(asserted, msg)  ((void) 0)

   instead?

Reply:
I felt like the lack of parentheses gave it more of a "fake keyword" feel rather than "fake function", but I can switch it.

You're definitely right about the ```((void) 0)``` suggestion; I'll fix that too.